### PR TITLE
Add docs and minor fixes

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +34,73 @@ composer require yiisoft/rbac-rules-container --prefer-dist
...
+        return $parameters['authorID'] === $userId;
+    }
+}
+
+$rulesContainer = new RulesContainer(new MyContainer());
+$rule = $rulesContainer->create(AuthorRule::class);
+```
+
+- `MyContainer` is a container for resolving dependencies and  must be an instance of 
+`Psr\Container\ContainerInterface`.  [Yii Dependency Injection](https://github.com/yiisoft/di) implementation also can
```
Comment (vjik) [inline README.md]:
```suggestion
`Psr\Container\ContainerInterface`. [Yii Dependency Injection](https://github.com/yiisoft/di) implementation also can 
```

File: README.md [hunk 2]
```diff
@@ -32,6 +34,73 @@ composer require yiisoft/rbac-rules-container --prefer-dist
...
+Basically, the arguments are the same as in [Yii Factory](https://github.com/yiisoft/factory). Please refer to its docs 
+for more details.
+
+Rules are created only once, then cached and reused for repeated calls.
+
+```php
+$rule = $rulesContainer->create(AuthorRule::class); // Returned from cache
+````
+
+### Using config
```
Comment (vjik) [inline README.md]: Need info that this config for config plugin.

File: README.md [hunk 3]
```diff
@@ -15,8 +15,10 @@
 [![static analysis](https://github.com/yiisoft/rbac-rules-container/workflows/static%20analysis/badge.svg)](https://github.com/yiisoft/rbac-rules-container/actions?query=workflow%3A%22static+analysis%22)
 [![type-coverage](https://shepherd.dev/github/yiisoft/rbac-rules-container/coverage.svg)](https://shepherd.dev/github/yiisoft/rbac-rules-container)
 
-The package provides rules container for [Yii RBAC (Role-Based Access Control)](https://github.com/yiisoft/rbac) package
-based on [Yii Factory](https://github.com/yiisoft/factory).
+This package is a factory for creating [Yii RBAC (Role-Based Access Control)](https://github.com/yiisoft/rbac) rules. It 
+provides rules container wrapping [Yii Factory](https://github.com/yiisoft/factory) and uses 
+[Yii Definitions](https://github.com/yiisoft/definitions) syntax. RBAC manager considers rule as a name and parameters
+and unaware of its creation by design, delegating creation to rules container keeping responsibilities separation.
```
Comment (samdark) [inline README.md]:
```suggestion
This package is a factory for creating [Yii RBAC (Role-Based Access Control)](https://github.com/yiisoft/rbac) rules. It 
provides rules container based on [Yii Factory](https://github.com/yiisoft/factory) and uses 
[Yii Definitions](https://github.com/yiisoft/definitions) syntax. RBAC manager accepts rule as a name and parameters
and is unaware of its creation by design, delegating creation to rules container keeping responsibilities separation.
```

File: README.md (hunk 1 above)
Comment (samdark) [inline README.md]:
```suggestion
- `MyContainer` is a container for resolving dependencies and must be an instance of 
`Psr\Container\ContainerInterface`. [Yii Dependency Injection](https://github.com/yiisoft/di) implementation also can 
```

File: README.md (hunk 3 above)
Comment (arogachev) [reply README.md]: Fun fact about "based on" - that's how it was initially. :slightly_smiling_face:

File: README.md (hunk 1 above)
Comment (arogachev) [reply README.md]: Fixed in https://github.com/yiisoft/rbac-rules-container/pull/6#discussion_r845453019.

**Review by samdark — APPROVED** [review]

**Pull request merged**